### DISCUSSION TITLE
milvus 如何给向量添加 ttl

### DISCUSSION THREAD

**Discussion by ponponon** [milvus-io/milvus#33688] [Q&A and General discussion]
我希望向量可以像 redis 的 key 一样，设置一个 ttl 倒计时，超过这个值就自动删除

但是我不知道怎么做, 因为在谷歌上搜索不到对应的 api/文档

---

我想把 ttl 设置的向量级别，而不是集合或者分区级别，可以做到吗？

---

如果支持的话，是从哪个版本开始支持的

**Comment by yhmo** [comment]
从2.2版本开始
https://milvus.io/docs/v2.2.x/modify_collection.md#Modify-a-collection

ttl只能针对collection级别。
ttl受GC的影响，删除过期数据的时间并不精确。

**Reply by ponponon** [reply]
如何在 attu 上设置 collection 的 ttl

<img width="1439" alt="图片" src="https://github.com/milvus-io/milvus/assets/38725104/260cc738-7caa-4fc9-9970-df38458ecdb8">

**Reply by ponponon** [reply]
> 从2.2版本开始 https://milvus.io/docs/v2.2.x/modify_collection.md#Modify-a-collection
> 
> ttl只能针对collection级别。 ttl受GC的影响，删除过期数据的时间并不精确。

还有一个事情，ttl 的取值范围是多少？文档上也没有写

**Reply by yhmo** [reply]
大于等于0即可，没有上限

**Reply by ponponon** [reply]
> 大于等于0即可，没有上限

没有上限不会出现整数溢出、浮点数溢出等问题吗？比如我设置一个 1000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 的 ttl 值
